src/test/java/guitests/PanelRenameTest.java

```diff
@@ -130,4 +137,22 @@ public void panelRenameTest() {
         push(KeyCode.ENTER);
         sleep(EVENT_DELAY);
     }
+
+    @Test
+    public void panelRenameSelectOnlyOnePanel() {
+        // Test for the currently selected panel when trying to rename
+
+        pushKeys(MAXIMIZE_WINDOW);
+        while (panels.getPanelCount() < 3) {
+            pushKeys(CREATE_RIGHT_PANEL);
```

Comment:
This is fragile. The `waitOnFxThread` inside `pushKeys` is not good enough as a guarantee for synchronisation, so `panels.getPanelCount` may execute for the next iteration of the loop before the keypress is even registered.

As a result, sometimes there will be 4-5 panels instead of just 3 panels, which causes this test (as well as the one above it) to break.